diff --git a/connectivity/drivers/emac/TARGET_STM/stm32xx_emac.cpp b/connectivity/drivers/emac/TARGET_STM/stm32xx_emac.cpp
index 0e7d64ad34..686cf69122 100644
--- a/connectivity/drivers/emac/TARGET_STM/stm32xx_emac.cpp
+++ b/connectivity/drivers/emac/TARGET_STM/stm32xx_emac.cpp
@@ -157,6 +157,9 @@ bool _phy_init()
     /* Set PHY IO functions */
     LAN8742_RegisterBusIO(&LAN8742, &LAN8742_IOCtx);
 
+    /* slo-eth */
+    LAN8742_SetLinkState(&LAN8742, LAN8742_STATUS_10MBITS_HALFDUPLEX);
+
     /* Initialize the LAN8742 ETH PHY */
     return LAN8742_Init(&LAN8742) == LAN8742_STATUS_OK;
 }
